File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        return getPrimes(Integer.toString(n, k));
    }
    public int getPrimes(String num) {
        int answer = 0;
        for(String s : num.split("0")) {
            if(!s.equals("") && isPrime(Long.parseLong(s))) answer++;
        }
        return answer;
    }
    public boolean isPrime(long num) {
        if(num == 1) return false;
        if(num == 2) return true;
        for(int i = 2 ; i <= Math.sqrt(num) ; i++) {
            if(num % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def convert(n, k):
    ans = ''
    while n > 0:
        ans += str(n % k)
        n //= k
    return ans[::-1]

def isPrime(n):
    if n == 1: return False
    for i in range(2, int(n ** 0.5 + 1)):
        if n % i == 0: return False
    return True

def solution(n, k): return sum(1 for s in convert(n, k).split('0') if s != '' and isPrime(int(s)))
    
    